src/App.tsx

```diff
@@ -26,25 +27,56 @@ const App = () => {
     const [arrivalStation, setArrivalStation] = React.useState('');
 
     const onSubmit = () => {
-        departureStation && arrivalStation && window.open(urlMaker(departureStation, arrivalStation));
+        departureStation
+        && arrivalStation
+        && window.open(urlMaker(departureStation, arrivalStation));
+    };
+
+    const disableSubmit = () => {
+        return !departureStation || !arrivalStation || departureStation===arrivalStation;
+    };
+
+    const getDisabledMessage = () => {
+        return <div className = "disable-message has-text-danger" >
+            {
+                (!departureStation || !arrivalStation)
+            && <>Select two stations.</>
+            || <>Departure station cannot be the same as arrival station.</>
+            }
+        </div>;
     };
 
     return <BrowserRouter>
         <div className = "App">
             <div className = "dropdown-menus-container">
-                <div className = "dropdown">
-                    <Dropdown valueUpdateFunction = { setDepartureStation } placeHolder = "Select Station" label = 'Departure:' selectableStations = { stationNames } id = 'departure-station-selection'/>
-                </div>
-                <div className = "dropdown">
-                    <Dropdown valueUpdateFunction = { setArrivalStation } placeHolder = "Select Station" label = 'Arrival:' selectableStations = { stationNames.filter((s) => s!==departureStation) } id = 'arrival-station-selection'/>
-                </div>
+                <StationDropdown
+                    valueUpdateFunction = { setDepartureStation }
+                    label = 'Departure:'
+                    selectableStations = { stationNames }
+                    id = 'departure-station-selection'
+                />
+                <StationDropdown
+                    valueUpdateFunction = { setArrivalStation }
+                    label = 'Arrival:'
+                    selectableStations = { stationNames }
+                    id = 'arrival-station-selection'
+                />
             </div>
-            
-            <Button text = 'Select Station' onClick = { onSubmit } classes = 'is-danger' disabled = { !departureStation || !arrivalStation }/>
+
+            {
+                disableSubmit() && getDisabledMessage()
```

Comment:
Do we want to be calling DisableSubmit() twice (line 67 and 74)?
Would this make more sense as a State?

Reply:
Fixed this :) made the `disableSubmit` boolean stateful.